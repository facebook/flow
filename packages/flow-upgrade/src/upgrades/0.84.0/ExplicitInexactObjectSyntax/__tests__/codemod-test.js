/**
 * Copyright (c) 2013-present, Facebook, Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @format
 * @flow
 */

'use strict';

const path = require('path');
const fs = require('fs');
const cp = require('child_process');
const jscodeshift = require('jscodeshift').withParser('flow');
const transform = require('../codemod');

const FIXTURES_DIR = path.join(__dirname, './fixtures');

fs.readdirSync(FIXTURES_DIR).forEach(fixture => {
  test(fixture, () => {
    const file = path.join(FIXTURES_DIR, fixture);
    const source = fs.readFileSync(file, 'utf8');
    const root = jscodeshift(source);
    const skipped = !transform(jscodeshift, root);
    expect(skipped ? null : root.toSource()).toMatchSnapshot();
  });
});
