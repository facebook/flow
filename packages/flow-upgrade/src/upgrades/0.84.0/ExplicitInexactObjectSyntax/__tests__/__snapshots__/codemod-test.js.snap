// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test.js 1`] = `
"//@flow
type T = {...};
opaque type T: {...} = {...};

function test(x: {...}): {...} {}
class A {}
interface B {}
declare class C {}
declare interface D {}

type Nested = { x: {...}, ... };
type Exact = {| |};

type Generic<T: {...}> = T;
opaque type Generic<T: {...}> = T;

class A {
  static x: {...} = {}
}

type T = {
  a: number,
  b: string,
  c: {...},
  d: string,
  ...
};
type U = {
  a: number,
  b: {...},
  c: { x : {...}, ... },
  d: { x: { y :{...}, ... }, ... },
  ...
};

// We should also make sure objects already using this syntax aren't broken by the codemod
type T = {...};
opaque type T: {...} = {...};

function test(x: {...}): {...} {}
class A {}
interface B {}
declare class C {}
declare interface D {}

type Nested = { x: {...}, ... };
type Exact = {| |};

type Generic<T: {...}> = T;
opaque type Generic<T: {...}> = T;

class A {
  static x: {...} = {}
}

type T = {
  a: number,
  b: string,
  c: {...},
  d: string,
  ...
};
type U = {
  a: number,
  b: {...},
  c: { x : {...}, ... },
  d: { x: { y :{...}, ... }, ... },
  ...
};
"
`;
